helpers/updateSubmission.ts

```diff
@@ -1,120 +1,125 @@
-import db from './dbload'
-import { SubmissionData } from '../@types/submission'
+import type { Challenge, Submission, User } from '.prisma/client'
+import {
+  MutationAcceptSubmissionArgs,
+  MutationRejectSubmissionArgs,
+  SubmissionStatus
+} from '../graphql'
+import { prisma } from '../prisma'
 import {
   getUserByEmail,
   publicChannelMessage,
   sendDirectMessage
 } from './mattermost'
-import { SubmissionStatus } from '../graphql'
-
-const { Submission, Challenge, User, UserLesson, Lesson } = db
 
-export type ArgsUpdateSubmission = {
-  id: number
-  lessonId: number
-  comment: string
+type ArgsUpdateSubmission = (
+  | MutationAcceptSubmissionArgs
+  | MutationRejectSubmissionArgs
+) & {
+  reviewerId: number
+  status: SubmissionStatus
 }
 
 export type ArgsChatNotification = {
-  submission: SubmissionData
+  submission: Submission & { challenge: Challenge; reviewer: User | null }
   userChatId: string
   reviewerId: number
 }
 
 export const updateSubmission = async (
-  args: ArgsUpdateSubmission & { reviewerId: number; status: SubmissionStatus }
-) => {
+  args: ArgsUpdateSubmission
+): Promise<Submission> => {
   try {
     if (!args) throw new Error('Invalid args')
-    const { id, comment, status, reviewerId } = args
-    // query submission that is being updated
-    const submission = await Submission.findByPk(id)
-
-    // count challenges inside lesson that submission belongs to and query user email
-    const [lessonChallengeCount, { email }] = await Promise.all([
-      Challenge.count({
-        where: { lessonId: submission.lessonId }
-      }),
-      User.findByPk(submission.userId)
-    ])
-
-    // update and save submission data
-    submission.set('reviewerId', reviewerId)
-    submission.set('status', status)
-    submission.set('comment', comment)
-    await submission.save()
+    const { id, comment, status, reviewerId, lessonId } = args
+    const { user, lesson, ...submission } = await prisma.submission.update({
+      where: { id },
+      data: {
+        reviewerId,
+        status,
+        comment
+      },
+      include: {
+        user: true,
+        reviewer: true,
+        challenge: true,
+        lesson: true
+      }
+    })
 
     const [
-      { id: userChatId, username: chatUsername },
-      lessonSubmissions
+      lessonChallengeCount,
+      passedLessonSubmissions,
+      userLesson,
+      { id: userChatId, username: chatUsername }
     ] = await Promise.all([
```

Comment:
Do we still need `Promise.all`? According to wiki it was a weird [quirk](https://github.com/garageScript/c0d3-app/wiki/Sequelize-Query-Performance) of sequelize.

Reply:
Dunno, what's bad about it?

I don't think it's really a "quirk", it's more of a way to start multiple processes in parallel.

Reply:
With sequelize `Promise.all` was much faster than one big query. 
We should delete this wiki page once sequelize is refactored.